monitor.c

```diff
@@ -301,7 +301,7 @@ monitor_child_preauth(struct ssh *ssh, struct monitor *pmonitor)
 		auth2_authctxt_reset_info(authctxt);
 
 		authenticated = (monitor_read(ssh, pmonitor,
-		    mon_dispatch, &ent) == 1);
+		    mon_dispatch, &ent, 1) == 1);
```

Comment:
Instead of doing like this, take dependency on the "privsep_auth_child".
the logic should go inside monitor_read_log(). 
Inside #ifdef windows block, you can do an "extern int privsep_auth_child", if privsep_auth_child is not true then add the "[pre-auth]" tag